stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -69,6 +70,18 @@ internal class PaymentSheetViewModel internal constructor(
     internal val _viewState = MutableLiveData<ViewState.PaymentSheet>(null)
     internal val viewState: LiveData<ViewState.PaymentSheet> = _viewState.distinctUntilChanged()
 
+    var checkoutIdentifier: CheckoutIdentifier? = null
+    internal fun getViewStateObservable(checkoutIdentifier: CheckoutIdentifier): MediatorLiveData<ViewState.PaymentSheet> {
+        val outputLiveData: MediatorLiveData<ViewState.PaymentSheet> =
+            MediatorLiveData<ViewState.PaymentSheet>()
+        outputLiveData.addSource(_viewState) { currentValue ->
+            if (this.checkoutIdentifier == checkoutIdentifier) {
+                outputLiveData.value = currentValue
+            }
+        }
+        return outputLiveData
+    }
```

Comment:
If I understand what you're trying to do, I think you can just do:

```
    var currCheckoutIdentifier: CheckoutIdentifier? = null

    internal fun getViewStateObservable(
        checkoutIdentifier: CheckoutIdentifier
    ) = liveData {
        emitSource(
            _viewState.map {
                if (currCheckoutIdentifier == checkoutIdentifier) {
                    it
                } else {
                    null
                }
            }
        )
    }
```

Reply:
I was following the pattern of distinctUntilChanged.  Are the benefits to this other approach?

Reply:
What you have is totally fine. Thought there was a way to make it slightly less verbose. Not sure what I suggested is really more concise though. :)

Reply:
ok, I am going to leave it